### ISSUE TITLE
xunit.runner.visualstudio.testadapter still on .NET 3.5

### ISSUE THREAD

**Issue by memark** [xunit/xunit#697]
In [_visualstudio.xunit_](https://github.com/xunit/visualstudio.xunit) I have noticed that the [_xunit.runner.visualstudio.testadapter_](https://github.com/xunit/visualstudio.xunit/blob/master/xunit.runner.visualstudio.testadapter/xunit.runner.visualstudio.testadapter.csproj) project is  targetting .NET Framework 3.5. This has the following implication:

If your assembly targets .NET 4.0 or newer, it's not possible to use _vstest.console.exe_, with just Visual Studio 2015 installed on the machine, . If you try, you get a popup telling you to install .NET 3.5, followed by

`Error: The active Test Run was aborted because the execution process exited unexpectedly. Check the execution process logs for more information. If the logs are
 not enabled, then enable the logs and try again.`

This in turn leads to that an out-of-the-box Visual Studio Online (hosted TFS) project cannot run xUnit tests without tweaking. The following error is encountered:

`Warning: Using Isolation mode to run tests as required by effective Platform:X86 and .Net Framework:Framework35 settings for test run. Use the /inIsolation parameter to suppress this warning.
 Warning: [xUnit.net 00:00:00.1600266] Exception discovering tests from MyProject.Tests: System.BadImageFormatException: Could not load file or assembly 'C:\a\1\s\MyProject.Tests\bin\Debug\xunit.execution.desktop.dll' or one of its dependencies. This assembly is built by a runtime newer than the currently loaded runtime and cannot be loaded.`
 
I'm assuming this targetting is for compatibility reasons? Are there any plans for retargetting towards 4.0?

**Comment by bradwilson** [comment]
The support matrix for this runner requires that we use 3.5 (it needs to support VS 2012, and it needs to support tests written in xUnit.net v1.x, which can link as far back as .NET 2.0).

**Comment by onovotny** [comment]
Can you manually specify a newer framework? I know that the VSO/VSTS hosted builds have VS 2010-2015 on them, so .NET 3.5 is definitely there. There may be a parameter needed to tell VS Test to use the 4.x version though.

**Comment by bradwilson** [comment]
I think passing `/Framework:Framework45` to `vstest.console.exe` should be sufficient.

**Comment by memark** [comment]
Thanks, that works! That's a good work-around, much better than the one I had found already. (I will update my newly created pull request with this solution.)

Just as a comment to the support matrix mentioned. VS 2012 was in fact shipped with .NET Framework 4.5, so assume that's not the reason for targetting .NET 3.5. The support for tests written in .NET 2.0 just might be though. It's been too long since I dealed with code that old.
